src/main/java/net/minecraftforge/common/BiomeManager.java

```diff
@@ -114,6 +119,44 @@ public static void removeBiome(BiomeType type, BiomeEntry entry)
         }
     }
 
+    /**
+     * Adds an OceanBiomeEntry to a type of ocean.
+     * This method should be called during mod loading to add a new OceanBiomeEntry to a type of ocean (e.g. a new Frozen Ocean biome).
+     * This method is threadsafe.
+     *
+     * @param type The type of ocean for this OceanBiomeEntry.
+     * @param entry The OceanBiomeEntry to add.
+     */
+    public static void addOceanBiome(OceanType type, OceanBiomeEntry entry)
+    {
+        int idx = type.ordinal();
+        synchronized (BiomeManager.class) {
+            List<OceanBiomeEntry> list = idx > oceanBiomes.length ? null : oceanBiomes[idx];
+            if (list != null) list.add(entry);
```

Comment:
Should this not be an error condition? When can the index be out of bounds so that list is null?

Reply:
Looking back at it, yes, it should really be an error condition.
Will fix 👍